packages/xo-web/src/xo-app/hub/recipes/recipe-form.js

```diff
@@ -100,6 +108,78 @@ export default decorate([
           value={value.nbNodes}
         />
       </FormGrid.Row>
+      <FormGrid.Row>
+        <label>
+          {_('recipeStaticIpAddresses')}
+          <input
+            className='form-control'
+            name='staticIpAddress'
+            onChange={effects.toggleStaticIpAddress}
+            type='checkbox'
+            value={value.staticIpAddress}
+          />
+        </label>
+      </FormGrid.Row>
+      {value.staticIpAddress && (
+        <FormGrid.Row>
+          <label>{_('recipeMasterIpAddress')}</label>
+          <input
+            className='form-control'
+            name='masterIpAddress'
+            onChange={effects.onChangeValue}
+            placeholder={formatMessage(messages.recipeMasterIpAddress)}
+            required
+            type='text'
+            value={value.masterIpAddress}
+          />
+        </FormGrid.Row>
+      )}
+      {value.staticIpAddress && (
+        <FormGrid.Row>
+          <label>{_('recipeNetworkMask')}</label>
+          <input
+            className='form-control'
+            name='networkMask'
+            onChange={effects.onChangeValue}
+            placeholder={formatMessage(messages.recipeNetworkMask)}
+            required
+            type='text'
+            value={value.networkMask}
+          />
+        </FormGrid.Row>
+      )}
+      {value.staticIpAddress && (
+        <FormGrid.Row>
+          <label>{_('recipeGatewayIpAddress')}</label>
+          <input
+            className='form-control'
+            name='gatewayIpAddress'
+            onChange={effects.onChangeValue}
+            placeholder={formatMessage(messages.recipeGatewayIpAddress)}
+            required
+            type='text'
+            value={value.gatewayIpAddress}
+          />
+        </FormGrid.Row>
+      )}
+      {value.staticIpAddress &&
+        [...Array(parseInt(value.nbNodes)).keys()].map(i => {
```

Comment:
We usually use the `+` operator instead of `parseInt`. IIRC, it's safer in some edge cases? :thinking:  And you can pass the array directly and use the 2nd argument in `map`:
```js
[...Array(+value.nbNodes))].map((_, i) => {
  // ...
})
```

Reply:
Thanks for the tip on the ```+``` operator, I didn't know it!

Reply:
What's the error? It seems to work for me when I do this:
```js
[...Array(5)].map((_, i) => {
  console.log(i)
})
```

Reply:
I got ```Uncaught TypeError: e is not a function```.

I'll investigate why and I'll keep you posted.

Reply:
I think the ```_``` line 166 causes the problem.